metals/src/main/scala/scala/meta/internal/metals/Command.scala

```diff
@@ -1,20 +1,82 @@
 package scala.meta.internal.metals
 
+import scala.reflect.ClassTag
+import scala.util.matching.Regex
+
 import scala.meta.internal.metals.MetalsEnrichments._
 
 import com.google.gson.JsonPrimitive
 import org.eclipse.{lsp4j => l}
 
+trait BaseCommand {
+  def id: String
+  def title: String
+  def description: String
+  def arguments: String
+
+  def toLSP(arguments: List[AnyRef]): l.Command =
+    new l.Command(title, id, arguments.asJava)
+
+  protected def isApplicableCommand(params: l.ExecuteCommandParams): Boolean = {
+    val command = Option(params.getCommand).getOrElse("")
```

Comment:
Is Option in `Option(params.getCommand)` needed if `getCommand` has `@Pure @NonNull` annotations?

Reply:
ach, it might not be needed then, I just copied it over.

Reply:
Simplified it to not use option.